src/lib.rs

```diff
@@ -390,58 +409,65 @@ pub trait Trait: system::Trait + Sized {
 }
 
 decl_storage! {
-    // A macro for the Storage trait, and its implementation, for this module.
-    // This allows for type-safe usage of the Substrate storage database, so you can
-    // keep things around between blocks.
-    trait Store for Module<T: Trait> as Example {
-        // Any storage declarations of the form:
-        //   `pub? Name get(getter_name)? [config()|config(myname)] [build(|_| {...})] : <type> (= <new_default_value>)?;`
-        // where `<type>` is either:
-        //   - `Type` (a basic value item); or
-        //   - `map KeyType => ValueType` (a map item).
-        //
-        // Note that there are two optional modifiers for the storage type declaration.
-        // - `Foo: Option<u32>`:
-        //   - `Foo::put(1); Foo::get()` returns `Some(1)`;
-        //   - `Foo::kill(); Foo::get()` returns `None`.
-        // - `Foo: u32`:
-        //   - `Foo::put(1); Foo::get()` returns `1`;
-        //   - `Foo::kill(); Foo::get()` returns `0` (u32::default()).
-        // e.g. Foo: u32;
-        // e.g. pub Bar get(bar): map T::AccountId => Vec<(BalanceOf<T>, u64)>;
-        //
-        // For basic value items, you'll get a type which implements
-        // `support::StorageValue`. For map items, you'll get a type which
-        // implements `support::StorageMap`.
-        //
-        // If they have a getter (`get(getter_name)`), then your module will come
-        // equipped with `fn getter_name() -> Type` for basic value items or
-        // `fn getter_name(key: KeyType) -> ValueType` for map items.
-        Dummy get(dummy) config(): Option<BalanceOf<T>>;
+    trait Store for Module<T: Trait> as Forum {
+
+        /// Map category identifier to corresponding category.
+        pub CategoryById get(category_by_id) config(): map CategoryId => Category<T::BlockNumber, T::Moment, T::AccountId>;
+
+        /// Category identifier value to be used for the next Category created.
+        pub NextCategoryId get(next_category_id) config(): CategoryId;
+
+        /// Map thread identifier to corresponding thread.
+        pub ThreadById get(thread_by_id): map ThreadId => Thread<T::BlockNumber, T::Moment, T::AccountId>;
+
+        /// Thread identifier value to be used for next Thread in threadById.
+        pub NextThreadId get(next_thread_id) config(): ThreadId;
 
-        // A map that has enumerable entries.
-        Bar get(bar) config(): linked_map T::AccountId => BalanceOf<T>;
+        /// Map post identifier to corresponding post.
+        pub PostById get(post_by_id): map PostId => Post<T::BlockNumber, T::Moment, T::AccountId>;
 
-        // this one uses the default, we'll demonstrate the usage of 'mutate' API.
-        Foo get(foo) config(): BalanceOf<T>;
+        /// Post identifier value to be used for for next post created.
+        pub NextPostId get(next_post_id) config(): PostId;
 
+        /// Account of forum sudo.
+        pub ForumSudo get(forum_sudo) config(): T::AccountId;
```

Comment:
Should it be an Option as we discussed?
```suggestion
        pub ForumSudo get(forum_sudo) config(): Option<T::AccountId>;
```

Reply:
See https://github.com/Joystream/substrate-forum-module/pull/3#discussion_r289319452